### ISSUE TITLE
Segfault on tracing

### ISSUE THREAD

**Issue by cpcdoy** [apitrace/apitrace#474]
Hi,

A segfault happens on this [project](https://github.com/cpcdoy/Engine) when tracing and calling glBufferData. I'm using the apitrace from the official ubuntu repository (16.04, so it must be version 6.1 not the latest version, but reporting anyway because it's on the official ubuntu repo).

I'm using `Mesa DRI Intel(R) HD Graphics 5600 (Broadwell GT2)` with OpenGL 4.3.

Here's the backtrace:
```
apitrace: warning: caught signal 11
apitrace: ignoring exception while tracing
/usr/bin/../lib/x86_64-linux-gnu/apitrace/wrappers/glxtrace.so+0x21401c
/lib/x86_64-linux-gnu/libc.so.6+0x3549f
/lib/x86_64-linux-gnu/libc.so.6+0x14d986
/usr/bin/../lib/x86_64-linux-gnu/apitrace/wrappers/glxtrace.so+0x216b8e
/usr/bin/../lib/x86_64-linux-gnu/apitrace/wrappers/glxtrace.so+0x210f3a
/usr/bin/../lib/x86_64-linux-gnu/apitrace/wrappers/glxtrace.so: glBufferData+0xb9
./Engine_check: _ZN14render_backend14opengl_backend12generate_vaoESt10shared_ptrIN8resource7gl_meshEE+0x2a3: /home/cpcdoy/projects/Engine/src/render_backend/opengl_backend.cc:131
./Engine_check: _ZN14render_backend14opengl_backend24generate_compatible_meshESt10shared_ptrIN8resource4meshEE+0x5b: /home/cpcdoy/projects/Engine/src/render_backend/opengl_backend.cc:71
./Engine_check: _ZN14render_backend14render_manager24generate_compatible_meshESt10shared_ptrIN8resource4meshEE+0x5e: /home/cpcdoy/projects/Engine/src/render_backend/render_manager.cc:38
./Engine_check: _ZN8resource16resource_manager9load_meshENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE+0x226: /home/cpcdoy/projects/Engine/src/resource/resource_manager.cc:39
./Engine_check: _ZN6engine6engine9load_meshENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE+0x6e: /home/cpcdoy/projects/Engine/src/engine/engine.cc:68
./Engine_check: main+0x11f: /home/cpcdoy/projects/Engine/tests/debug/main.cc:15
/lib/x86_64-linux-gnu/libc.so.6: __libc_start_main+0xef
./Engine_check: _start+0x28
?
apitrace: info: taking default action for signal 11

**Comment by jrfonseca** [comment]
@cpcdoy Could you please build apitrace from source and confirm the issue repros with latest master? If not, I'm afraid Ubuntu packages are beyond my responsibility.

**Comment by cpcdoy** [comment]
Well, somehow after an upgrade of my ubuntu (which included going from OpenGL 4.3 to 4.4) I cannot reproduce this behavior anymore even with the same apitrace...

**Comment by jrfonseca** [comment]
> Could it have come from Intel's driver or even Mesa ?

Yes, it's probably the case.  apitrace relies on OpenGL driver answers from `glGet*` calls to decide what/how much data to record.

I'm gonna close for now, but if you hit again with newer drivers/apitrace feel free to reopen.
